services/src/main/scala/cromwell/pipeline/service/ProjectConfigurationService.scala

```diff
@@ -40,24 +62,40 @@ object ProjectConfigurationService {
       def addConfiguration(projectConfiguration: ProjectConfiguration, userId: UserId): Future[Unit] =
         projectService
           .getUserProjectById(projectConfiguration.projectId, userId)
-          .flatMap(_ => repository.addConfiguration(projectConfiguration))
+          .recoverWith { case e: ProjectServiceException => serviceErrorMapper(e) }
+          .flatMap(
+            _ =>
+              repository.addConfiguration(projectConfiguration).recoverWith {
+                case _ => internalError("add configuration")
+              }
+          )
 
       private def getByProjectId(projectId: ProjectId, userId: UserId): Future[Seq[ProjectConfiguration]] =
-        projectService.getUserProjectById(projectId, userId).flatMap(_ => repository.getAllByProjectId(projectId))
-
-      def getLastByProjectId(projectId: ProjectId, userId: UserId): Future[Option[ProjectConfiguration]] =
-        getByProjectId(projectId, userId).map(
-          _.filter(_.active).sortBy(_.version).lastOption
-        )
+        projectService
+          .getUserProjectById(projectId, userId)
+          .recoverWith { case e: ProjectServiceException => serviceErrorMapper(e) }
+          .flatMap(
+            _ => repository.getAllByProjectId(projectId).recoverWith { case _ => internalError("find configuration") }
+          )
+      private def getLastOptionByProjectId(projectId: ProjectId, userId: UserId): Future[Option[ProjectConfiguration]] =
+        getByProjectId(projectId, userId).map(_.filter(_.active).sortBy(_.version).lastOption)
+
+      def getLastByProjectId(projectId: ProjectId, userId: UserId): Future[ProjectConfiguration] =
+        getLastOptionByProjectId(projectId, userId).flatMap {
+          case Some(config) => Future.successful(config)
+          case None         => Future.failed(NotFound())
+        }
 
       def deactivateLastByProjectId(projectId: ProjectId, userId: UserId): Future[Unit] =
-        getLastByProjectId(projectId: ProjectId, userId: UserId).flatMap {
+        getLastOptionByProjectId(projectId: ProjectId, userId: UserId).flatMap {
           case Some(config) => updateConfiguration(config.copy(active = false))
-          case _            => Future.failed(new RuntimeException("There is no project to deactivate"))
+          case None         => internalError("found configuration")
```

Comment:
Why is this an internal error? If we couldn't find a configuration because it doesn't exist or user doesn't have access to it, it's not our internal error. It is a valid business error.

Reply:
Done.